sources/packages/backend/apps/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -80,6 +82,37 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
     private readonly cancelAssessmentQueue: Queue<CancelAssessmentQueueInDTO>,
   ) {
     super(dataSource.getRepository(EducationProgramOffering));
+    this.applicationOfferingChangeRepo = dataSource.getRepository(
+      ApplicationOfferingChangeRequest,
+    );
+  }
+
+  /** Validates student authorization for the given education program offering.
+   * @param offeringId offering id.
+   * @param studentId student id.
+   * @returns true if the student is authorized for the given offering, otherwise false.
+   */
+  async validateApplicationOfferingForStudent(
+    offeringId: number,
+    studentId: number,
+  ): Promise<boolean> {
+    const applicationOfferingChangeId =
+      this.applicationOfferingChangeRepo.findOne({
+        select: { id: true },
+        relations: { application: { student: true } },
+        where: [
+          {
+            application: { student: { id: studentId } },
+            activeOffering: { id: offeringId },
+          },
+          {
+            application: { student: { id: studentId } },
+            requestedOffering: { id: offeringId },
+          },
+        ],
+      });
+    if (applicationOfferingChangeId) return true;
```

Comment:
pls use `{}`

Reply:
Changed it to `!!applicationOfferingChangeId`